addons/hearing/functions/fnc_explosion.sqf

```diff
@@ -0,0 +1,44 @@
+#include "..\script_component.hpp"
+/*
+ * Author: johnb43
+ * Handles deafness due to explosions going off near the player.
+ *
+ * Arguments:
+ * 0: Projectile <OBJECT>
+ * 1: Explosion position ASL <ARRAY>
+ * 2: Velocity <ARRAY> (unused)
+ *
+ * Return Value:
+ * None
+ *
+ * Example:
+ * [_projectile, [0, 0, 0], [0, 0, 0]] call ace_hearing_fnc_explosion
+ *
+ * Public: No
+ */
+
+params ["_projectile", "_pos"];
+
+// Don't allow for distances under 1
+private _distance = ((eyePos ACE_player) vectorDistance _pos) max 1;
```

Comment:
small optimization to ignore far away shots if we're adding this to all explosions, 
just guessing at what the cutoff should be

and block ear ringing if player in spectator (should match previous behavior)

```suggestion
private _distance = ((eyePos ACE_player) vectorDistance _pos) max 1;
if (_distance > 100) exitWith {}; // fast exit if explosion far away
if ((getNumber (configOf ACE_player>> "isPlayableLogic")) == 1) exitWith {};  
```